### DISCUSSION TITLE
Sections for Migrations / Tests would also be useful

### DISCUSSION THREAD

**Discussion by ghost** [olivierlacan/keep-a-changelog#479] [General]

**Comment by olivierlacan** [comment]
@harsha-mudi I don't think I understand what you mean. Could you elaborate?

**Comment by ghost** [comment]
- Each version should:
  - List its release date in the above format.
  - Group changes to describe their impact on the project, as follows:
    - `Added` for new features.
    - `Changed` for changes in existing functionality.
    - `Deprecated` for once-stable features removed in upcoming releases.
    - `Removed` for deprecated features removed in this release.
    - `Fixed` for any bug fixes.
    - `Security` to invite users to upgrade in case of vulnerabilities.

I have found it useful to summarise database migrations in a separate section.
Testing too could deserve its own section !

**Comment by olivierlacan** [comment]
I don't think testing changes belong in a change log entry. I don't believe changes to tests are relevant to end-users. As for migrations, if they correspond to `Added` or `Changed` functionality, the existing categories already cover that use case.

Shields.io already offers ways for project maintainers to report test coverage with a badge: 
[![Coveralls](https://img.shields.io/coveralls/jekyll/jekyll.svg)]()

The evolution of test coverage at a high level (significant percentage changes) could be a relevant change but once again that's something which overlaps quite well with the `Changed` section.

Example:

``` markdown
## Added
- Specs to cover critical business logic that was previously untested

## Changed
- Sporadically failing tests by stubbing some calls.
```

**Comment by lunemec** [comment]
I thought that migrations would fall into the area of upgrading to next version, which as discussed here https://github.com/olivierlacan/keep-a-changelog/issues/34 should be in separate file with instructions how to upgrade to next version.

**Comment by ghost** [comment]
@lunemec, @olivierlacan 

Actually this issue is moot if one can use custom headings.
In #51 there is a mention of a tag called internal. Whats about  simple headings ?

> Some projects also use HISTORY.txt, HISTORY.md, History.md, NEWS.txt, NEWS.md, News.txt, RELEASES.txt, RELEASE.md, releases.md, etc.

Indeed.

I suppose I just want to throw everything inside the Changelog.  
The end result kinda looking like https://www.freebsd.org/releases/10.1R/relnotes.html.

I maybe conflating two different ideas though !

**Comment by Ajedi32** [comment]
See also #30
